src/table.js

```diff
@@ -45,17 +40,53 @@ export function Table(
     if (iterindex >= 0) {
       iterindex = iterator = undefined;
       index = Uint32Array.from(array = arrayify(data), (_, i) => i);
+      N = index.length;
     }
   }
 
+  let next;
+
+  function minlengthof(length) {
+    if (N != null) {
+      return Math.min(N, length);
+    }
+
+    if (next == null) {
+      next = iterator.next;
+    }
+
+    if (length <= iterindex) {
+      return length;
+    }
+
+    while (length > iterindex) {
+      const {done, value} = next();
```

Comment:
```suggestion
      const {done, value} = iterator.next();
```

Can we do this instead of stashing `next`?

Reply:
Done